File: go/src/classes/classes.go
package main

import "fmt"
import "math"

type Vertex struct {
	X float64
	Y float64
}

func (v Vertex) length() float64 {
	return math.Sqrt(
		math.Pow(v.X, 2.0) +
		math.Pow(v.Y, 2.0),
	)
}

func (v Vertex) normalize() {
	length := v.length()
	v.X /= length
	v.Y /= length
}

func (v *Vertex) zero() {
	v.X = 0.0
	v.Y = 0.0
}

func main() {
	v := Vertex{X: 6, Y: 7}
	fmt.Printf("Length: %.3f\n", v.length())
	// Length: 9.220

	v.normalize()
	fmt.Println("Normalized:", v.X, v.Y)
	// Normalized: 6 7

	v.zero()
	fmt.Println("Zeroed:", v.X, v.Y)
	// Zeroed: 0 0
}

File: js/src/classes/classes.js
function Vertex(x, y) {
  this.x = x;
  this.y = y;
}

Vertex.prototype.length = function() {
  return Math.sqrt(
      Math.pow(this.x, 2) +
      Math.pow(this.y, 2)
  );
};

Vertex.prototype.normalize = function() {
  var length = this.length();
  this.x /= length;
  this.y /= length;
};

Vertex.prototype.zero = function() {
  this.x = 0;
  this.y = 0;
};

function main () {
  var v = new Vertex(6, 7);
  console.log("Length:", v.length().toFixed(3));
  // Length: 9.220

  v.normalize();
  console.log("Normalized:", v.x, v.y);
  // Normalized: 0.651 0.759

  v.zero();
  console.log("Zeroed:", v.x, v.y);
  // Zeroed: 0 0
}
main();